src/blockchains/eth_generator/consts.ts

```diff
@@ -1,21 +1,24 @@
 import { IDataType } from "../../IOracleData";
 
 export type ETHType = 'uint' | 'uint8' | 'int' | 'string' | 'price'
-export interface IETHDataType
+interface IDTStub
 {
     name: string
     hash: string
     type: ETHType
-    value: string
     decimals?: number
 }
-export interface IWideDataType extends IETHDataType { value: any; life: number; update: number }
+export interface IETHDataType extends IDTStub
+{
+    value: string
+}
+export interface IWideDataType extends IDTStub { [key:string]: string|number|boolean|undefined|null, value: string|number|boolean; life: number; update: number }
 export const evaluator = (d: IWideDataType) => {
     switch (d.type) {
-        case 'price': return `Price(${d.value}, ${d.decimals})`
+        case 'price': return !!d.value ? `Price(${d.value}, ${!!d.decimals ? d.decimals : 0})` : PUSH_CONSTRUCTION[d.type].default
```

Comment:
Intention not clear: better use string interpolation in both cases with `d.value || PUSH_CONSTRUCTION[d.type].default`
If `d.value` was passed, `d.decimals` MUST be passed as well.